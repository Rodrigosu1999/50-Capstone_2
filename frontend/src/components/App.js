import { BrowserRouter } from "react-router-dom";
import "../styles/App.css";
import Routes from "./Routes";
import NavBar from "./NavBar";
import ContextProvider from "./ContextProvider";

//Our main App, here we are passing our context and rendering our routes and navbar 
function App() {

  return (
    <div className="App">
      <ContextProvider>
      <BrowserRouter>
        <NavBar />
        <main>
          <Routes />
        </main>
      </BrowserRouter>
      </ContextProvider>
    </div>
  );
}

export default App;
