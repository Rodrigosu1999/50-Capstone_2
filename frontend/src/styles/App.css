main {
  width: 100vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 2vh 5vh;
  height: calc(100vh - 56px); /* router view - navbar height */
  overflow: scroll;
  color: rgb(0, 0, 0);
}

.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
}

.Spinner{
  position: absolute;
  top: 50%;
  left: 50%;
}