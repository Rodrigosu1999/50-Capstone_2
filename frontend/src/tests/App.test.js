import React from "react";
import { fireEvent, render } from '@testing-library/react';
import App from '../components/App';

it("renders without crashing", function () {
  render(<App/>);
});

it("render matches snapshot", function () {
  const {asFragment} = render(<App/>);
  expect(asFragment()).toMatchSnapshot();
});


it("works when you click on the login and register button", function() {
  const { queryByText } = render(<App />);

  // expect the first window to show 
  expect(queryByText("The place to find your next destination")).toBeInTheDocument();
  expect(queryByText("World")).not.toBeInTheDocument();

  // move forward to register form
  const registerButton = queryByText("Register");
  fireEvent.click(registerButton);

  // expect register form to show
  expect(queryByText("Username")).toBeInTheDocument();
  expect(queryByText("Password")).toBeInTheDocument();
  expect(queryByText("First Name")).toBeInTheDocument();
  expect(queryByText("Last Name")).toBeInTheDocument();
  expect(queryByText("Email")).toBeInTheDocument();
  expect(queryByText("World")).not.toBeInTheDocument();

  // move forward to login form
  const loginButton = queryByText("Login");
  fireEvent.click(loginButton);

  // expect login form to show
  expect(queryByText("Username")).toBeInTheDocument();
  expect(queryByText("Password")).toBeInTheDocument();
  expect(queryByText("Email")).not.toBeInTheDocument();
  expect(queryByText("World")).not.toBeInTheDocument();
});

it("works when you set values in the login inputs", function() {
  const { queryByText, queryByLabelText } = render(<App />);

  // expect login form to show 
  expect(queryByText("Username")).toBeInTheDocument();
  expect(queryByText("Password")).toBeInTheDocument();
  expect(queryByText("Email")).not.toBeInTheDocument();
  expect(queryByText("World")).not.toBeInTheDocument();

  const usernameInput = queryByLabelText("Username");
  fireEvent.change(usernameInput, {target: {value: "test"}});
  expect(usernameInput.value).toBe('test');
  
  const passwordInput = queryByLabelText("Password");
  fireEvent.change(passwordInput, {target: {value: "password"}});
  expect(passwordInput.value).toBe('password');
});
