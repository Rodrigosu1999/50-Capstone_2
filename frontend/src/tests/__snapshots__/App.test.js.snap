// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <nav
        class="navbar navbar-expand-md"
      >
        <div
          class="container-fluid"
        >
          <a
            class="navbar-brand"
          >
            <img
              alt="logo"
              src="globe_logo.png"
            />
          </a>
          <span>
            Please use the pages buttons and logout when you finish
          </span>
          <ul
            class="ms-auto nav"
          >
            <li
              class="nav-item"
            >
              <a
                href="/login"
                id="nav-login"
              >
                Login
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                href="/register"
                id="nav-register"
              >
                Register
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
    <main>
      <div
        class="Welcome Welcome-Anon"
      >
        <div
          class="Welcome-Anon-card card"
        >
          <div
            class="text-center card-body"
          >
            <div
              class="card-title"
            >
              <h3
                class="font-weight-bold"
              >
                Culture Quests
              </h3>
            </div>
            <div
              class="card-subtitle"
            >
              <h4>
                The place to find your next destination
              </h4>
              <h4>
                And learn about countries
              </h4>
            </div>
            <button
              class="btn btn-primary"
              type="button"
            >
              Login!
            </button>
            <button
              class="btn btn-primary"
              type="button"
            >
              Register!
            </button>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
